### PR TITLE
update error msg for invalid pool info

### PR DESCRIPTION
**Pull request by nanli1**
#  When pool info not return the value that need to be checked, we should not return 'KEY ERROR' 
* virsh_pool: update error msg for invalid pool info

##
- [ ] Test results
   avocado run --vt-type libvirt --vt-machine-type q35 virsh.pool.positive_test.pool_type_scsi --vt-connect-uri qemu:///system
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : 41b9ed3f147c521b005326f6ddda057e92ad93f5
JOB LOG    : /root/avocado/job-results/job-2021-12-02T05.42-41b9ed3/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.pool.positive_test.pool_type_scsi: **FAIL: The value Capacity you checked is not returned in pool_info : {'Name': 'virsh_pool_test', 'UUID': 'ebb03fab-1339-4593-8975-978d7035eb4b', 'State': 'inactive', 'Persistent': 'yes', 'Autostart': 'no'} (21.75 s)**
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 22.27 s

### CODE REVIEW

File: libvirt/tests/src/virsh_cmd/pool/virsh_pool.py [hunk 1]
```diff
@@ -156,22 +156,26 @@ def check_pool_info(pool_info, check_point, value):
...
-            # exceed 1 percent.
-            if is_in_range(float(pool_info[check_point].split()[0]),
-                           float(value.split()[0]), 1):
-                logging.debug("Pool '%s' is '%s'.", check_point, value)
-            else:
-                test.fail("Pool '%s' isn't '%s'." %
-                          (check_point, value))
+        # If pool info not return check Key value pair , raise appropriate error message
+        if pool_info.get(check_point) is None:
+            test.fail("The value {} you checked is not returned in pool_info : {}".format(check_point, pool_info))
```
Comment (Yingshun) [inline libvirt/tests/src/virsh_cmd/pool/virsh_pool.py:161]: Could you please limit all lines to a maximum of 79 characters?

File: libvirt/tests/src/virsh_cmd/pool/virsh_pool.py [hunk 2]
```diff
@@ -156,22 +156,26 @@ def check_pool_info(pool_info, check_point, value):
...
-        if check_point in val_tup and float(value.split()[0]):
-            # As from bytes to GiB, could cause deviation, and it should not
-            # exceed 1 percent.
-            if is_in_range(float(pool_info[check_point].split()[0]),
-                           float(value.split()[0]), 1):
-                logging.debug("Pool '%s' is '%s'.", check_point, value)
-            else:
-                test.fail("Pool '%s' isn't '%s'." %
-                          (check_point, value))
+        # If pool info not return check Key value pair , raise appropriate error message
```
Comment (Yingshun) [inline libvirt/tests/src/virsh_cmd/pool/virsh_pool.py:159]: Could you please update the comment here? I don't understand what 'check Key value pair' is here.

File: libvirt/tests/src/virsh_cmd/pool/virsh_pool.py [hunk 3]
```diff
@@ -370,7 +374,10 @@ def check_pool_info(pool_info, check_point, value):
                 new_info = _pool.pool_info(pool_name)
                 check_items = ["Capacity", "Allocation", "Available"]
                 for i in check_items:
-                    check_pool_info(pool_info, i, new_info[i])
+                    if new_info.get(i) is None:
+                        test.fail("The value {} you checked is not returned in pool_info : {}".format(i, new_info))
```
Comment (Yingshun) [inline libvirt/tests/src/virsh_cmd/pool/virsh_pool.py:378]: Please help limit all lines to a maximum of 79 characters...

**Comment by Yingshun** [comment]
BTW, @nanli1 could you please update the commit message, so others can understand the purpose of this PR. Thank you!

File: libvirt/tests/src/virsh_cmd/pool/virsh_pool.py (hunk 1 above)
Comment (nanli1) [reply libvirt/tests/src/virsh_cmd/pool/virsh_pool.py]: done

File: libvirt/tests/src/virsh_cmd/pool/virsh_pool.py (hunk 2 above)
Comment (nanli1) [reply libvirt/tests/src/virsh_cmd/pool/virsh_pool.py]: done

File: libvirt/tests/src/virsh_cmd/pool/virsh_pool.py (hunk 3 above)
Comment (nanli1) [reply libvirt/tests/src/virsh_cmd/pool/virsh_pool.py]: done

**Comment by nanli1** [comment]
> BTW, @nanli1 could you please update the commit message, so others can understand the purpose of this PR. Thank you!

done

**Review by Yingshun — APPROVED** [review]

**Review by chunfuwen — APPROVED** [review]
LGTM

File: libvirt/tests/src/virsh_cmd/pool/virsh_pool.py [hunk 4]
```diff
@@ -156,22 +156,28 @@ def check_pool_info(pool_info, check_point, value):
...
-        if check_point in val_tup and float(value.split()[0]):
-            # As from bytes to GiB, could cause deviation, and it should not
-            # exceed 1 percent.
-            if is_in_range(float(pool_info[check_point].split()[0]),
-                           float(value.split()[0]), 1):
-                logging.debug("Pool '%s' is '%s'.", check_point, value)
-            else:
-                test.fail("Pool '%s' isn't '%s'." %
-                          (check_point, value))
+        # If pool info not returns the value you need to check,
```
Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/pool/virsh_pool.py:159]: It is better to say `' If pool info does not return the value you need to check'`

**Review by dzhengfy — COMMENTED** [review]
Codes LGTM, but usually for a PR to fix an issue, we pasted the log w/o the fix and with the fix. Thanks.

File: libvirt/tests/src/virsh_cmd/pool/virsh_pool.py (hunk 4 above)
Comment (nanli1) [reply libvirt/tests/src/virsh_cmd/pool/virsh_pool.py]: done

**Review by dzhengfy — APPROVED** [review]
A reminder for naming your PR:
A PR to fix a problem, please add <changed-file>: fix <issue short desc> on PR title
A PR to automate a case, please add <changed/added-file>: add case <case short desc> on PR title
@nanli1

**Comment by nanli1** [comment]
> A reminder for naming your PR: A PR to fix a problem, please add : fix on PR title A PR to automate a case, please add <changed/added-file>: add case on PR title @nanli1

Got it, Thanks for reminding

**Pull request merged**